In [ ]:

!pip install ipython-sql
!pip install sqlalchemy
!pip install psycopg2
!pip install sql
import sqlite3

In [2]:
from sqlalchemy.engine import create_engine
import sqlite3
from pandas.io import sql
import subprocess
%reload_ext sql

In [3]:
conn = sqlite3.connect("C:\\Users\\mansi\\Downloads\\international_debt.db")

In [ ]:
%sql sqlite:///C:/Users/mansi/Downloads/international_debt.db

## 1.The World Bank's international debt data

It's not that we only humans take debts to manage our necessities. A country may also take debt to manage its economy.For example, infrastructure spending is one costly ingridient required for the countrys, citizens to lead comfortable lives.The World Bank is the organization that provides debt to countries.

In this notebook, we are going to analyze the internationsl debt data collected by The World Bank. The dataset contains information about the amount of debt(in USD) owed by developing countries across several catagories. We are going to find answers to questions like:


  1.What is the total amount of debt that is owed by the countries listed in the dataset?
  2.Which country owns the maximum amount of debt and what does that amount look like?
  3.What is the average amount of debt owed by the countries across different debt           indicators?

  
Let's first SELECT all of the columns from the international_debt.Also,we'll limit the output to the first ten rows to keep the output clean.
  

In [ ]:
%%sql
SELECT * FROM international_debt LIMIT 10;

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', 72894453.7),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', 53239440.1),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', 61739336.9),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', 49114729.4),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', 39903620.1),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', 39107845),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', 23779724.3),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', 13335820),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', 100847181.9),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', 72894453.7)]

## 2.Finding the number of distinct countries

From the 1st 10 rows, we can see the amount of debt owed by Afganisthan in the different debt indicators. But we do not know the number of different countries we have on the tabel.
There are repititon in the country names because a country is most likely to have debt in more that one debt indicator

Without a count of unique countries we will not be able to perform our statistical analysis holistically.In this section, we are going to extract the number of unique countries in the table.

In [ ]:
%%sql
SELECT
    COUNT(DISTINCT country_name) AS total_distinct_countries
FROM international_debt;

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[(124,)]

## 3.Finding out the distinct debt indicators 

We can see there are total of 124 countries present on the table. As we saw in the first section, there is a column called indicator_name that briefly specifies the purpoe of taking the debt. Just beside that column, there is another column called indicator_code which symbolizes the category of these debts. Knowing about these various debt indicators will help us to understand the areas in which a country can possibly indebted to.

In [ ]:
%%sql
SELECT DISTINCT indicator_code AS distinct_debt_indicators
FROM international_debt
ORDER BY distinct_debt_indicators;

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',),
 ('DT.AMT.PBND.CD',),
 ('DT.AMT.PCBK.CD',),
 ('DT.AMT.PROP.CD',),
 ('DT.AMT.PRVT.CD',),
 ('DT.DIS.BLAT.CD',),
 ('DT.DIS.DLXF.CD',),
 ('DT.DIS.MLAT.CD',),
 ('DT.DIS.OFFT.CD',),
 ('DT.DIS.PCBK.CD',),
 ('DT.DIS.PROP.CD',),
 ('DT.DIS.PRVT.CD',),
 ('DT.INT.BLAT.CD',),
 ('DT.INT.DLXF.CD',),
 ('DT.INT.DPNG.CD',),
 ('DT.INT.MLAT.CD',),
 ('DT.INT.OFFT.CD',),
 ('DT.INT.PBND.CD',),
 ('DT.INT.PCBK.CD',),
 ('DT.INT.PROP.CD',),
 ('DT.INT.PRVT.CD',)]

## Totaling the ammount of debt owned by the countries

As mentioned earlier, the financial debt of a particular company represents its economic state.But if we were to project this on an overall global scale, how will we approach it?

Lets's switch gears from debt indicators now and find out the total amount of debt(in USD) that is owned by the different countries. This will give us a sense of how the overall economy of the entire world is holding up.

In [ ]:
%%sql
SELECT
    ROUND(SUM(debt)/1000000, 2) AS total_debt
FROM international_debt

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[(3079734.49,)]

## 5.Country with the highest debt

"Human beings cannot comprehend very large or very small numbers. It would be useful for us to acknowledge that fact" -Daniel Kahneman. That is more than 3 million millon USD, an amount which is really hard for us to fanthom.

Now we have the exact total of the amounts of debt owned by several countries, lets now find out the country that owns the highest amount of debt along with the amount. *NOTE* that this debt is the sum of the different debts owned by a country across several categories. This will help us to understand more about the country in terms of socio- economic scenarios. We can also find out the category in which the country owns its highest debt. But we will leave that for now.

In [ ]:
%%sql
SELECT
    country_name,
    SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 1;

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[('China', 285793494734.2)]

## 6. Average amount of debt across indicators

So, it was China. A more in-depth breakdown of China's debts can be found here.

We now have a brief overview of the dataset and a few of its summary statistics. We already have an idea of different debt indicators in which the countries owe their debts. We can dig even further to find out of an average how much debt a country owes? This will give us a better sense of the distribution of the amount of debt across different indicators

In [ ]:
%%sql
SELECT
    indicator_code AS debt_indicator,
    indicator_name,
    AVG(debt) AS average_debt
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt DESC
LIMIT 10

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', 5904868401.499195),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', 5161194333.812659),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', 2152041216.8902435),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', 1958983452.8598356),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', 1803694101.963265),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', 1644024067.6508067),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', 1223139290.39823),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', 1220410844.4215188),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', 1191187963.083064),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', 1082623947.6536236)]

## 7.The highest amount of principal repayments

We can see that the indicator DT.AMT.DLFX.CD tops the charts of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital. More info about this category can be found here.

An interesting observation in the above findings is that there is a huge difference in the amounts of the indicators after the second one. This indicates that there is a huge difference in the amounts of the indicators after the second one.This indicates that the 1st 2 indicators might be the most severe categories in which the countries owe their debts.

We can investigate this a bit more so as to find out which country owes the highest amount of debts in the category of long term debt (DT.AMT.DLFX.CD). Since not all countries suffer from the same kind of ecnomic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically

In [40]:
%%sql 
SELECT 
    country_code,
    indicator_name
FROM international_debt
WHERE debt = (SELECT
               MAX(debt)
             FROM international_debt
             WHERE indicator_code = 'DT.AMT.DLXF.CD');

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[('CHN', 'Principal repayments on external debt, long-term (AMT, current US$)')]

## 8.The most common debt indicator

China has the highest amount of debt in the long-term debt(DT.AMT.DLXF.CD)category. This is verified by the World Bank. It is often a good idea to verify our analyses like this since it validates that our investigation is correct.

We saw that long-term is the top-most category when it comes o the average amount of debt. But is it the most common indicator in which the countries owe their debt? Lets find that out.

In [38]:
%%sql 
SELECT indicator_code,
COUNT(indicator_code) AS indicator_count
FROM international_debt
GROUP BY indicator_code
ORDER BY indicator_count DESC, indicator_code DESC
LIMIT 20;

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69)]

## 9. Other viable debt issues and conclusions

There are a total of six debt indicators in which all countries listed in our dataset have taken debt. The indicator DT.AMT.DLFX.CD is also there in the list. So, this gives us a clue that all these countries are suffering from a common economic issue.But that is not the end of the story, but just a part of the story.

Let's change tracts from debt_indicator now and focus on the amount of debt again. Lets find out tha maximum amount of debt that each country has. Withj this, we willl be in a position to identify the other plausible economic issue a country might be going through.

In this, notebook we took a look at debt owed by countries across the globe. We extracted a few summary statistics from the data and unraveled some interesting facts and figures. We also validated out findings to make sure the investigations our correct.

In [52]:
%%sql
SELECT country_name, MAX(debt) AS maximum_debt
FROM international_debt
GROUP BY country_name
ORDER BY maximum_debt DESC
LIMIT 10;

 * sqlite:///C:/Users/mansi/Downloads/international_debt.db
Done.


[('China', 96218620835.7),
 ('Brazil', 90041840304.1),
 ('Russian Federation', 66589761833.5),
 ('Turkey', 51555031005.80001),
 ('South Asia', 48756295898.19999),
 ('Least developed countries: UN classification', 40160766261.6),
 ('IDA only', 34531188113.2),
 ('India', 31923507000.8),
 ('Indonesia', 30916112653.8),
 ('Kazakhstan', 27482093686.4)]